### DISCUSSION TITLE
Loading / saving content

### DISCUSSION THREAD

**Discussion by dlemmermann** [gluonhq/rich-text-area#105] [Q&A]
Could somebody please provide information on how the rich text area can be populated with content via API and also the other round ... retrieving the content so that it can be persisted? Maybe a WIKI page?

**Comment by jperedadnr** [comment]
So far, the way to add rich content to the control is by providing a `FaceModel` object, like in the Main demo class, that can be currently accessed via File -> Open Text.

<img width="465" alt="image" src="https://user-images.githubusercontent.com/2043230/157919541-bbc36e7b-56e3-4b28-ae84-5885c1dcd321.png">

The API isn't nice yet, you need to create the decorations for each fragment of the text, like:

```
List<DecorationModel> decorations = List.of(
                new DecorationModel(0, 12, TextDecoration.builder().presets().build()),
                new DecorationModel(12, 3, TextDecoration.builder().presets().fontWeight(FontWeight.BOLD).build()),
                new DecorationModel(15, 1, new ImageDecoration(Main.class.getResource("gluon_logo-150x150@2x.png").toURI().toString(), 32, 32)),
                new DecorationModel(16, 11, TextDecoration.builder().presets().fontPosture(FontPosture.ITALIC).build()),
                new DecorationModel(27, 15, TextDecoration.builder().presets().foreground(Color.RED).build())
        );

FaceModel faceModel = new FaceModel("Simple text one\u200b two three\nExtra line text", decorations, 42);
```

This should be simplified.

So far, the faceModel is not updated back with the changes done, so that should be fixed next.
